src/tests/ftest/container/snapshot_aggregation.py

```diff
@@ -0,0 +1,159 @@
+#!/usr/bin/python
+"""
+  (C) Copyright 2020 Intel Corporation.
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+
+  GOVERNMENT LICENSE RIGHTS-OPEN SOURCE SOFTWARE
+  The Government's rights to use, modify, reproduce, release, perform, display,
+  or disclose this software are subject to the terms of the Apache License as
+  provided in Contract No. B609815.
+  Any reproduction of computer software, computer software documentation, or
+  portions thereof marked with this legend must also reproduce the markings.
+"""
+import time
+
+from ior_test_base import IorTestBase
+from daos_utils import DaosCommand
+
+
+class SnapshotAggregation(IorTestBase):
+    # pylint: disable=too-many-ancestors
+    """Defines snapshot aggregation test cases.
+
+    Test Class Description:
+          Verify snapshot aggregation.
+
+    :avocado: recursive
+    """
+
+    def __init__(self, *args, **kwargs):
+        """Initialize a SnapshotAggregation object."""
+        super(SnapshotAggregation, self).__init__(*args, **kwargs)
+        self.dmg = None
+        self.free_space = {"scm": [], "nvme": []}
+
+    def update_free_space(self):
+        """Append the free space list with the current pool capacities."""
+        for index, name in enumerate(("scm", "nvme")):
+            self.free_space[name].append({
+                "dmg": self.pool.query_data[name]["free"],
+                "api": int(self.pool.info.pi_space.ps_space.s_free[index])
+            })
+
+    def detect_aggregation_complete(self):
+        """Detect that aggregation has started and completed."""
+        # Detect aggregation start
+        self.log.info("Waiting for 90 seconds for aggregation to start")
+        time.sleep(90)
+
+        # Detect aggregation completion
+        self.log.info("Waiting for 30 seconds for aggregation to finish")
+        time.sleep(30)
```

Comment:
I would recommend looping in smaller chunks rather just going to sleep in one go for 120 secs.
Something like 60 secs of loop and check back and if not complete do it again. You may decide how many times you would like to loop. 2-3 seems reasonable to me.
The aggregation completion time here is somewhat hardcoded. And this value can change in future due to either performance improvement or degradation. I believe if the aggregation finishes in 60 secs we will be wasting 1 minute on all runs for no reason. But if it doesn't finish in 60 secs we can loop back and check it again and if still doesn't finish we can fail the test.

Reply:
Done.